crates/scroll/alloy/network/src/lib.rs

```diff
@@ -0,0 +1,228 @@
+        self.as_mut().set_input(input);
+    }
+
+    fn from(&self) -> Option<Address> {
+        self.as_ref().from()
+    }
+
+    fn set_from(&mut self, from: Address) {
+        self.as_mut().set_from(from);
+    }
+
+    fn kind(&self) -> Option<TxKind> {
+        self.as_ref().kind()
+    }
+
+    fn clear_kind(&mut self) {
+        self.as_mut().clear_kind();
+    }
+
+    fn set_kind(&mut self, kind: TxKind) {
+        self.as_mut().set_kind(kind);
+    }
+
+    fn value(&self) -> Option<U256> {
+        self.as_ref().value()
+    }
+
+    fn set_value(&mut self, value: U256) {
+        self.as_mut().set_value(value);
+    }
+
+    fn gas_price(&self) -> Option<u128> {
+        self.as_ref().gas_price()
+    }
+
+    fn set_gas_price(&mut self, gas_price: u128) {
+        self.as_mut().set_gas_price(gas_price);
+    }
+
+    fn max_fee_per_gas(&self) -> Option<u128> {
+        self.as_ref().max_fee_per_gas()
+    }
+
+    fn set_max_fee_per_gas(&mut self, max_fee_per_gas: u128) {
+        self.as_mut().set_max_fee_per_gas(max_fee_per_gas);
+    }
+
+    fn max_priority_fee_per_gas(&self) -> Option<u128> {
+        self.as_ref().max_priority_fee_per_gas()
+    }
+
+    fn set_max_priority_fee_per_gas(&mut self, max_priority_fee_per_gas: u128) {
+        self.as_mut().set_max_priority_fee_per_gas(max_priority_fee_per_gas);
+    }
+
+    fn gas_limit(&self) -> Option<u64> {
+        self.as_ref().gas_limit()
+    }
+
+    fn set_gas_limit(&mut self, gas_limit: u64) {
+        self.as_mut().set_gas_limit(gas_limit);
+    }
+
+    fn access_list(&self) -> Option<&AccessList> {
+        self.as_ref().access_list()
+    }
+
+    fn set_access_list(&mut self, access_list: AccessList) {
+        self.as_mut().set_access_list(access_list);
+    }
+
+    fn complete_type(&self, ty: ScrollTxType) -> Result<(), Vec<&'static str>> {
+        match ty {
+            ScrollTxType::L1Message => Err(vec!["not implemented for deposit tx"]),
+            _ => {
+                let ty = TxType::try_from(ty as u8).unwrap();
+                self.as_ref().complete_type(ty)
+            }
+        }
+    }
+
+    fn can_submit(&self) -> bool {
+        self.as_ref().can_submit()
+    }
+
+    fn can_build(&self) -> bool {
+        self.as_ref().can_build()
+    }
+
+    #[doc(alias = "output_transaction_type")]
+    fn output_tx_type(&self) -> ScrollTxType {
+        match self.as_ref().preferred_type() {
+            TxType::Eip1559 | TxType::Eip4844 => ScrollTxType::Eip1559,
+            TxType::Eip2930 => ScrollTxType::Eip2930,
+            TxType::Legacy => ScrollTxType::Legacy,
+            TxType::Eip7702 => unimplemented!(),
+        }
+    }
+
+    #[doc(alias = "output_transaction_type_checked")]
+    fn output_tx_type_checked(&self) -> Option<ScrollTxType> {
+        self.as_ref().buildable_type().map(|tx_ty| match tx_ty {
+            TxType::Eip1559 | TxType::Eip4844 => ScrollTxType::Eip1559,
```

Comment:
Why do we cast `Eip4844` to `Eip1559`?

Reply:
Following Op to do it. I think it's mainly because we don't support Eip4844, so let eip4844 use eip1559 to check it.